packages/teleterm/src/services/config/configService.ts

```diff
@@ -1,39 +1,180 @@
-import { merge } from 'lodash';
+import { z } from 'zod';
 
-import { keyboardShortcutsConfigProvider } from './providers/keyboardShortcutsConfigProvider';
-import { appearanceConfigProvider } from './providers/appearanceConfigProvider';
-import { Config, ConfigService, ConfigServiceProvider } from './types';
+import { FileStorage } from 'teleterm/services/fileStorage';
+import { Platform } from 'teleterm/mainProcess/types';
 
-type ConfigServiceProviders = {
-  readonly [Property in keyof Config]: ConfigServiceProvider<Config[Property]>;
-};
+import { createConfigStore } from './configStore';
 
-export class ConfigServiceImpl implements ConfigService {
-  private config: Config;
-  private configProviders: ConfigServiceProviders = {
-    keyboardShortcuts: keyboardShortcutsConfigProvider,
-    appearance: appearanceConfigProvider,
-  };
+const appConfigSchema = z.object({
+  'keymap.tab-1': z.string(),
+  'keymap.tab-2': z.string(),
+  'keymap.tab-3': z.string(),
+  'keymap.tab-4': z.string(),
+  'keymap.tab-5': z.string(),
+  'keymap.tab-6': z.string(),
+  'keymap.tab-7': z.string(),
+  'keymap.tab-8': z.string(),
+  'keymap.tab-9': z.string(),
+  'keymap.tab-close': z.string(),
+  'keymap.tab-new': z.string(),
+  'keymap.tab-previous': z.string(),
+  'keymap.tab-next': z.string(),
+  'keymap.toggle-connections': z.string(),
+  'keymap.toggle-clusters': z.string(),
+  'keymap.toggle-identity': z.string(),
+  'keymap.open-quick-input': z.string(),
+  sansSerifFontFamily: z.string(),
+  monoFontFamily: z.string(),
```

Comment:
TBH, I'm not sure if we want to make fonts and shortcuts configurable right now.

AFAIK we don't validate those strings in any way which means that it opens us to an attack vector where arbitrary CSS can be injected through the config file. And I'm not sure if there are any attack vectors related to keyboard shortcuts.

Reply:
Yeah, a better validation is needed. I disabled reading them from the file.